test/sagemaker_tests/mxnet/training/__init__.py

```diff
@@ -11,3 +11,39 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 from __future__ import absolute_import
+
+import sagemaker
+from sagemaker.mxnet.estimator import MXNet as MXNet_origin
+
+
+class MXNet(MXNet_origin):
+    def __init__(self, image_uri, sagemaker_regions, **kwargs):
+        from ... import get_sagemaker_session
+        super().__init__(image_uri=image_uri, **kwargs)
+        self.image_uri = image_uri
+        self.sagemaker_regions = sagemaker_regions
+        self.sagemaker_session = get_sagemaker_session(self.sagemaker_regions[0])
+
+    def fit(self, inputs=None, wait=True, logs="All", job_name=None, experiment_config=None, **kwargs):
+        from ... import get_ecr_image_region, get_sagemaker_session, get_ecr_image
+
+        for region in self.sagemaker_regions:
+            self.sagemaker_session = get_sagemaker_session(region)
+            # Upload the image to test region if needed
+            if region != get_ecr_image_region(self.image_uri):
+                self.image_uri = get_ecr_image(self.image_uri, region)
+            try:
+                super().fit(
```

Comment:
Does this actually work? The `MXNET` estimator is initiated with ECR image and SM session to the first region in `sagemaker_regions` list. On a ICE, the `super.fit(...)` api is called again but the `MXNET` estimator is not created/updated with the new ECR image and SM session? Does initiating `self.image_uri` and `self.sagemaker_session` do the job?

Reply:
Correct. When you call `super.fit()` it uses current values for `self.image_uri` and `self.sagemaker_session`.